code/datums/components/nuclear_bomb_operator.dm

```diff
@@ -108,8 +108,11 @@
 	disky = WEAKREF(potential_disky)
 	mob_parent.update_appearance(updates = UPDATE_ICON)
 	mob_parent.balloon_alert(mob_parent, "disk secured!")
-	if (on_disk_collected)
-		on_disk_collected.Invoke(potential_disky)
+	INVOKE_ASYNC(src, PROC_REF(on_disk_collected), potential_disky)
```

Comment:
You don't need to do this, you can `on_disk_collected?.InvokeAsync(potential_disky)`